src/gui/editors/SongEditor.cpp

```diff
@@ -404,7 +480,68 @@ void SongEditor::closeEvent( QCloseEvent * ce )
 		hide();
 	}
 	ce->ignore();
- }
+}
+
+
+
+
+void SongEditor::mousePressEvent(QMouseEvent *_me)
+{
+	if (allowRubberband())
+	{
+		//we save the position of both scrollbars into a QPoint
+		m_scrollPos = QPoint(m_leftRightScroll->value(), contentWidget()->verticalScrollBar()->value());
+		//and the mouse position
+		m_origin = contentWidget()->mapFromParent(QPoint(_me->pos().x(), _me->pos().y()));
+		//and the zooming
+		m_currentZoomingValue = zoomingModel()->value();
+		//paint the rubberband
+		rubberBand()->setEnabled(true);
+		rubberBand()->setGeometry(QRect( m_origin, QSize()));
+		rubberBand()->show();
+
+		//the trackView(index) and the miditime where the mouse has clicked
+		const TrackView * tv = trackViewAt(_me->pos().y() - m_timeLine->height());
+		m_rubberBandStartTrackview = 0;
+		if (tv)
```

Comment:
A duplicate of line 350. Please extract it into something like `updateRubberbandSelectionForMousePos(pos)`.